### PR TITLE
Case importer datadog

### PR DESCRIPTION
**Pull request by dannyroberts**
- Create `datadog_gauge_task` for easily running a function
- use it to track total file size of case upload files over time

### CODE REVIEW

**Comment by dannyroberts** [comment]
I should see one of these for staging on datadog shortly after 5pm (in 40 minutes).

**Comment by dannyroberts** [comment]
Ah, I was reminded that staging isn't hooked up to datadog. I tested this in a production shell with copy-pasting, and I've separately tested that the periodic_task part of it works, so this is good to merge by me.

**Comment by dannyroberts** [comment]
(And no need to wait for tests)

**Comment by benrudolph** [comment]
sweet 🔢 🐶

**Comment by dannyroberts** [comment]
Weee! Still open to comments on the utility I added if there are any suggestions

**Comment by czue** [comment]
i like this a lot - looks like it could be a really useful tool in the toolkit

File: corehq/apps/case_importer/tasks.py [hunk 1]
```diff
@@ -279,3 +283,10 @@ def _submit_caseblocks(domain, case_type, caseblocks):
         'errors': errors.as_dict(),
         'num_chunks': num_chunks,
     }
+
+
+total_bytes = datadog_gauge_task(
+    'case_importer.files.total_bytes',
```
Comment (snopoke) [inline corehq/apps/case_importer/tasks.py]: have been trying to keep all metrics coming from commcare prefixed with `commcare` to make it easier to search in datadog.

File: corehq/util/datadog/gauges.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from functools import wraps
+from celery.task import periodic_task
+from corehq.util.datadog import statsd
+
+
+def datadog_gauge_task(name, fn, run_every):
+    """"
```
Comment (snopoke) [inline corehq/util/datadog/gauges.py]: nit: extra `"`

File: corehq/util/datadog/gauges.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+from functools import wraps
+from celery.task import periodic_task
+from corehq.util.datadog import statsd
+
+
+def datadog_gauge_task(name, fn, run_every):
```
Comment (snopoke) [inline corehq/util/datadog/gauges.py]: this is nice

File: corehq/apps/case_importer/tasks.py (hunk 1 above)
Comment (dannyroberts) [inline corehq/apps/case_importer/tasks.py]: Thanks, this is the kind of thing I knew I must have missed. Barely any data's come in so I'll just change this now. Out of curiosity

- is there any reason to restrict the length of these? I assume `'commcare.case_importer.files.total_bytes'` is fine, but if there's reason for it to be shorter, or there's some better convention to follow, I can use something else.
- There's no way to change the name of a metric already in datadog, right? I'm imagining a case where we'd already submitted a few months of data to this, and then decided we wanted it prefixed with `commcare`, but didn't want to lose continuity.

File: corehq/util/datadog/gauges.py (hunk 2 above)
Comment (dannyroberts) [inline corehq/util/datadog/gauges.py]: 🐦 👁

File: corehq/apps/case_importer/tasks.py (hunk 1 above)
Comment (snopoke) [inline corehq/apps/case_importer/tasks.py]: No reason to make them short that I know of. The way I think of these is that the dot prefixes are used to group related metrics.

Comment (dannyroberts) [inline corehq/apps/case_importer/tasks.py]: Cool, that's how I was thinking of it too.

**Pull request merged**